mailpoet/lib/Subscribers/ImportExport/PersonalDataExporters/NewslettersExporter.php

```diff
@@ -15,21 +16,31 @@ class NewslettersExporter {
   /** @var NewsletterUrl */
   private $newsletterUrl;
 
+  /*** @var SubscribersRepository */
+  private $subscribersRepository;
+
+  /*** @var NewslettersRepository */
+  private $newslettersRepository;
+
   public function __construct(
-    NewsletterUrl $newsletterUrl
+    NewsletterUrl $newsletterUrl,
+    SubscribersRepository $subscribersRepository,
+    NewslettersRepository $newslettersRepository
   ) {
     $this->newsletterUrl = $newsletterUrl;
+    $this->subscribersRepository = $subscribersRepository;
+    $this->newslettersRepository = $newslettersRepository;
   }
 
   public function export($email, $page = 1) {
-    $data = $this->exportSubscriber(Subscriber::findOne(trim($email)), $page);
+    $data = $this->exportSubscriber($this->subscribersRepository->findOneBy(['email' => trim($email)]), $page);
     return [
       'data' => $data,
       'done' => count($data) < self::LIMIT,
     ];
   }
 
-  private function exportSubscriber($subscriber, $page) {
+  private function exportSubscriber(SubscriberEntity $subscriber = null, $page) {
```

Comment:
The first parameter should not be optional but only nullable because it is always set.